transmogrifier/datacite.py

```diff
@@ -0,0 +1,187 @@
+        metadata record.
+    """
+    xml_element = ET.fromstring(datacite_xml_string)
+    kwargs = {}
+    titles_xml = xml_element.findall(".//datacite:title", NAMESPACES)
+    for title_xml in [t for t in titles_xml if "titleType" not in t.attrib]:
+        kwargs["title"] = title_xml.text
+
+    identifiers_xml = xml_element.findall(".//datacite:identifier", NAMESPACES)
+    for identifier_xml in [
+        i
+        for i in identifiers_xml
+        if "identifierType" in i.attrib and i.attrib["identifierType"] == "DOI"
+    ]:
+        doi = identifier_xml.text
+        kwargs["timdex_record_id"] = f"jpal:{doi.replace('/', '-')}"
+        kwargs["identifiers"] = [Identifier(value=doi, kind="DOI")]
+        kwargs["source_link"] = source_link_url + doi
+
+    contributors = create_instance_list_from_xml(
+        xml_element,
+        ".//datacite:contributor",
+        Contributor,
+        {
+            "kind": "contributorType",
+            "affiliation": "affiliation",
+            "identifier": "nameIdentifier",
+        },
+        "datacite:contributorName",
+    )
+    creators = create_instance_list_from_xml(
+        xml_element,
+        ".//datacite:creator",
+        Contributor,
+        {
+            "affiliation": "affiliation",
+            "identifier": "nameIdentifier",
+        },
+        "datacite:creatorName",
+    )
+    for creator in creators:
+        creator.kind = "Creator"
+    kwargs["contributors"] = contributors + creators
+
+    kwargs["publication_information"] = create_value_list_from_xml(
+        xml_element, ".//datacite:publisher"
+    )
+
+    dates = create_instance_list_from_xml(
+        xml_element,
+        ".//datacite:date",
+        Date,
+        {
+            "kind": "dateType",
+            "note": "dateInformation",
+        },
+    )
+    publication_years = []
+    publication_years_xml = xml_element.findall(
+        ".//datacite:publicationYear", NAMESPACES
+    )
+    for publication_year_xml in publication_years_xml:
+        publication_year = Date(value=publication_year_xml.text, kind="Publication")
+        publication_years.append(publication_year)
+    kwargs["dates"] = dates + publication_years
+
+    resource_types = []
+    content_types = []
+    resource_types_xml = xml_element.findall(".//datacite:resourceType", NAMESPACES)
+    for resource_type_xml in resource_types_xml:
+        if resource_type_xml.text is not None:
+            resource_types.append(
+                Note(value=resource_type_xml.text, kind="ResourceType")
+            )
+        if "resourceTypeGeneral" in resource_type_xml.attrib:
+            content_types.append(resource_type_xml.attrib["resourceTypeGeneral"])
+    kwargs["notes"] = resource_types
+    kwargs["content_type"] = content_types
+
+    kwargs = {k: v for k, v in kwargs.items() if v}
+    return kwargs
+
+
+def create_instance_list_from_xml(
+    xml_element: Any,
```

Comment:
Re: type hinting,  I know I’m “cheating” with the `Any`s but `defusedxml` doesn’t have an equivalent of `xml.etree.ElementTree.Element` and bandit barks if I import from `xml.etree.ElementTree `.